File: src/main/java/ru/nucodelabs/gem/app/io/JsonFileManager.java
package ru.nucodelabs.gem.app.io;

import java.io.File;
import java.io.Serializable;

public interface JsonFileManager {

    static JsonFileManager createDefault() {
        return new JsonFileManagerImpl();
    }

    <T extends Serializable> T loadFromJson(File jsonFile, Class<T> type) throws Exception;

    <T extends Serializable> void saveToJson(File jsonFile, T object) throws Exception;
}

File: src/main/java/ru/nucodelabs/gem/app/io/JsonFileManager.kt
package ru.nucodelabs.gem.app.io

import java.io.File

interface JsonFileManager {
    @Throws(Exception::class)
    fun <T> loadFromJson(jsonFile: File, type: Class<T>): T

    @Throws(Exception::class)
    fun <T> saveToJson(jsonFile: File, obj: T)
}